test/dsa_no_digest_size_test.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "testutil.h"
+
+#include <openssl/evp.h>
+#include <openssl/dsa.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+
+static DSA *dsa;
+
+static int genkeys(void)
+{
+    dsa = DSA_new();
+    if (!dsa) {
+        return 0;
+    }
+    if (!DSA_generate_parameters_ex(dsa, 2048, NULL, 0, NULL, NULL, NULL)) {
+        return 0;
+    }
+    if (!DSA_generate_key(dsa)) {
+        return 0;
+    }
+
+    return 1;
+}
+
+static int sign_and_verify(int len)
+{
+    /*
+     * Per FIPS 186-4, the hash is recommended to be the same length as q.
+     * If the hash is longer than q, the leftmost N bits are used; if the hash
+     * is shorter, then we left-pad (see appendix C.2.1).
+     */
+    size_t sigLength;
+    const BIGNUM *p, *q, *g;
+    int digestlen;
+    int ok = 0;
+
+    unsigned char *dataToSign = NULL;
+    unsigned char *paddedData = NULL;
+    unsigned char *signature = NULL;
+    EVP_PKEY_CTX *ctx = NULL;
+    EVP_PKEY *pkey = NULL;
+
+    DSA_get0_pqg(dsa, &p, &q, &g);
+    digestlen = BN_num_bytes(q);
+
+    dataToSign = OPENSSL_malloc(len);
+    paddedData = OPENSSL_malloc(digestlen);
+
+    if (!dataToSign || !paddedData) goto end;
```

Comment:
put the goto on a separate line.  there's a couple of other places where statements are joined, that we want on separate lines.